### PR TITLE
Db/auth

### PR DESCRIPTION
**Pull request by LukaszBielsko**
Authorisation and database added

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (pokiujf) [inline .vscode/launch.json]: Add this file to gitignore and uncommit it.
If you are not sure how to exclude a file from committed changes, (and will not find it in some tutorial) then we can make a call

Comment (LukaszBielsko) [reply .vscode/launch.json]: job done

File: src/containters/MainShop/MainShop.js [hunk 2]
```diff
@@ -1,88 +1,217 @@
...
+        const orders = this.state.orders.map((order) => {
+            if (order.orderID === orderID) {
+                order.status = 'realised'
+            }
+            return order
+        })
+        // create array with orders of given company
+        const companyOrders = []
+        orders.forEach(order => {
+            if (order.company === company) {
```
Comment (pokiujf) [inline src/containters/MainShop/MainShop.js]: You could do it with simpler with `const companyOrders = orders.find(order => order.company === company)`

File: src/components/Products/Products.js [hunk 3]
```diff
@@ -7,15 +7,19 @@ const Products = (props) => {
         <div>
             {props.products.map((product) => {
                 return <Product
-                    key={product.id}
+                    isAdmin={props.isAdmin}
+                    /* TODO does not work as a unique key */ 
+                    key={product.id} t
```
Comment (pokiujf) [inline src/components/Products/Products.js]: what is this ` t`?

File: src/components/Products/Product/Product.js [hunk 4]
```diff
@@ -35,53 +35,71 @@ class Product extends Component {
...
+                            <input
+                                value={howManyProducts.toString()}
+                                type="number"
+                                onChange={this.quantityInputHandler} >
+                            </input>
+                        </div>
+                        <p> Price for all items: {endPrice} </p>
+                        <button
+                            onClick={() => {
+                                // arrow function did not work here
```
Comment (pokiujf) [inline src/components/Products/Product/Product.js]: What do you mean?

File: src/components/InventoryPage/InventoryPage.js [hunk 5]
```diff
@@ -1,11 +1,129 @@
...
+
+    afterOpenModal = () => {
+        // references are now sync'd and can be accessed.
+        this.subtitle.style.color = 'green';
+    }
+
+    closeModal = () => {
+        this.setState({
+            modalIsOpen: false,
+            newProduct: {
```
Comment (pokiujf) [inline src/components/InventoryPage/InventoryPage.js]: Do we need to have this whole object? What about having just `newProduct: null` or `newProduct: {}`?

File: src/components/Products/Products.js (hunk 3 above)
Comment (LukaszBielsko) [reply src/components/Products/Products.js]: typo removed

File: src/containters/MainShop/MainShop.js (hunk 2 above)
Comment (LukaszBielsko) [reply src/containters/MainShop/MainShop.js]: I'm iterating over orders array that holds orders from all companies and I'm returning multiple orders from given company to a new array (companyOrders).
Array.find called on orders would return only first element, but there are more items to be returned.

File: src/components/OrdersPage/OrdersPage.js [hunk 6]
```diff
@@ -1,30 +1,31 @@
...
 
 class Orders extends Component {
 
-    showProps = () => console.log(this.props.orders.map((el) => el.order))
-
     render() {
+        const { orders, isAdmin, realiseOrder } = this.props;
 
-        let orders = this.props.orders.map((order, index) => {
+        let ordersList = orders.map((order) => {
```
[the remark this answers is not in the document]
Comment (LukaszBielsko) [reply src/components/OrdersPage/OrdersPage.js]: I've removed assigning order list to variable and moved orders.map directly to render method.

File: src/components/Products/Product/Product.js (hunk 4 above)
Comment (LukaszBielsko) [reply src/components/Products/Product/Product.js]: I put two method calls in a onClick handler. When they were called using fat arrow notation they did not work, so I left a comment for myself.

File: src/components/InventoryPage/InventoryPage.js (hunk 5 above)
Comment (LukaszBielsko) [reply src/components/InventoryPage/InventoryPage.js]: newProduct set to null when modal closes

File: src/components/Navigation/NavigationBar/NavigationBar.js [hunk 7]
```diff
@@ -10,21 +10,32 @@ let styles = {
...
-        <NavLink to='/orders' className={classes.navLink} activeStyle={styles}> Orders</NavLink>
-        <NavLink to='/inventory' className={classes.navLink} activeStyle={styles}> Inventory</NavLink>
 
-        {/* <Route path='/shop' component={ShopPage} />
-        <Route path='/cart' component={CartPage} />
-        <Route path='/orders' component={OrdersPage} />
-        <Route path='/inventory' component={InventoryPage} /> */}
+        {props.isLoggedIn ?
+            <>
+                {props.isAdmin ? null :
```
[the remark this answers is not in the document]
Comment (LukaszBielsko) [reply src/components/Navigation/NavigationBar/NavigationBar.js]: Changed conditional rendering in whole component to the advised one.

**Pull request merged**